core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3808,8 +3808,14 @@ class KafkaApis(val requestChannel: RequestChannel,
       )
   }
 
+  private def groupVersion(): GroupVersion = {
+    GroupVersion.fromFeatureLevel(metadataCache.features.finalizedFeatures.getOrDefault(GroupVersion.FEATURE_NAME, 0.toShort))
+  }
+
   private def isConsumerGroupProtocolEnabled(): Boolean = {
-    groupCoordinator.isNewGroupCoordinator && config.groupCoordinatorRebalanceProtocols.contains(Group.GroupType.CONSUMER)
+    groupCoordinator.isNewGroupCoordinator &&
```

Comment:
The GroupVersion will always increase like the MetadataVersion because we will use it to gate future changes related to groups in the future too (e.g. share groups). So we also have `groupCoordinatorRebalanceProtocols` as a static config to allow admin to disable the new protocol or other future protocols if they really don't want them. Honestly, I don't expect it to be really used but it is nice to have.

Reply:
Give me a bit of time to finish everything :). My next PR will change the default to `classic,consumer`. The plan is to have the new protocol enabled by default in 4.0. I will do this change separately though because it requires some cleaning too.